src/domains/field/compiler/resolver.ts

```diff
@@ -62,6 +62,7 @@ function computeFinalArgs(
 export function compileFieldResolver(
   target: Function,
   fieldName: string,
+  isGetter: boolean,
```

Comment:
Can it be replaced in favor of using `Object.getOwnPropertyDescriptor` to detect it automatically? Passing it over like this might make it harder to maintain

Reply:
it could, but doing `Object.getOwnPropertyDescriptor` is performance costly. This might sound like a micro optimisation, but it's not. See for example https://github.com/chaijs/chai/issues/127#issuecomment-12210969.

Having it detected only once is worth the faster performance IMHO. Decorators always run statically at startup, so it doesn't make sense to support a scenario where someone would change a property into a getter at runtime.